### ISSUE TITLE
mpileup with -l option is not an exact subset of mpileup without the -l option

### ISSUE THREAD

**Issue by stevendavis** [samtools/samtools#282]
When mpileup is run with -l listfile containing a list of positions, I was expecting the generated pileup to be an exact subset of the pileup generated without the -l option.  However, the generated pileup records at the first few specified positions are often different from the pileup records at those same positions when -l is not specified.

Below is an example showing position 1 missing completely when expected, and position 765 with a change in depth.

Without -l
-----
> seq       1       C       1       ^!G     F
> seq       765     T       10      .,.+3GTA,...,.. 9D0II:I3II

With -l
----
> seq       765     T       11      ,.,.+3GTA,...,..        F9D0II:I3II

**Comment by kirkmcclure** [comment]
This appears to be a documentation issue.

The samtools man page has no explanation of expected non-BED file format:
"-l, --positions	FILE	BED or position list file containing a list of regions or sites where pileup or BCF should be generated. If BED, positions are 0-based half-open [null]"

The description for bed_read() in bedidx.c contains:
   "The alternative format was originally for reading positions in VCF format.  This expects two columns, which indicate the reference and a position.  The position corresponds to a single base, and unlike BED counts from 1.

   "Which format is in use is determined based on whether one or two numbers can be decoded on the line.  As this choice is made line-by-line in this implementation, it is possible (but probably a bad idea) to mix both formats in the same file.  If trying to read a VCF file by this method, it would be important to ensure that the third column (ID) does not contain any entries that start with a digit, to avoid the line erroneously being parsed as a BED file entry."

Both 0.1.19 and 1.2 versions seem to handle both BED and (conforming) list files correctly.
````
./samtools view -S -b -o /tmp/view.001.bam test/dat/view.001.sam
bamtobed -i /tmp/view.001.bam > /tmp/view.001.bed
./samtools mpileup /tmp/view.001.bam > /tmp/view.001.mpileup
./samtools mpileup -l /tmp/view.001.bed /tmp/view.001.bam > /tmp/view.001.+l.mpileup
diff /tmp/view.001.mpileup /tmp/view.001.+l.mpileup

cat /tmp/view.001.bed | grep ref2 > /tmp/view.001.-ref1.bed
./samtools mpileup -l /tmp/view.001.-ref1.bed /tmp/view.001.bam > /tmp/view.001.+l.mpileup
diff /tmp/view.001.mpileup /tmp/view.001.+l.mpileup ew.001.-ref1.b
1,56d0
< ref1  1       N       0
< ref1  2       N       0
...
< ref1  55      N       1       g       8
< ref1  56      N       1       t$      8

cat /tmp/view.001.subset
ref1	1
ref1	2
ref1	3
ref1	4
ref1	5
ref2	16
ref2	31
ref2	46
./samtools mpileup -l /tmp/view.001.subset /tmp/view.001.bam > /tmp/view.001.+l.mpileup
diff /tmp/view.001.mpileup /tmp/view.001.+l.mpileup
6,71d5
< ref1  6       N       0
< ref1  7       N       0
...
< ref2  14      N       1       a       ~
< ref2  15      N       1       c$      ~
73,86d6
< ref2  17      N       1       t       }
< ref2  18      N       1       a       }
...
< ref2  29      N       1       c       }
< ref2  30      N       1       g$      }
88,101d7
< ref2  32      N       1       T       |
< ref2  33      N       1       G       |
...
< ref2  44      N       1       G       |
< ref2  45      N       1       A$      |
103,116d8
< ref2  47      N       1       T       {
< ref2  48      N       1       G       {
...
< ref2  59      N       1       G       {
< ref2  60      N       1       A$      {
````

**Comment by jmarshall** [comment]
I suspect this was another instance of the "BAQ not calculated at the first few bases of each chromosome" problem (#79), fixed by #407 and in the **1.3** release.
